astropy/table/table.py

```diff
@@ -436,6 +436,10 @@ def _init_from_list(self, data, names, dtype, n_cols, copy):
         def_names = _auto_names(n_cols)
 
         for col, name, def_name, dtype in zip(data, names, def_names, dtype):
+            # Structured ndarray gets viewed as a mixin
+            if isinstance(col, np.ndarray) and len(col.dtype) > 1:
+                col = col.view(NdarrayMixin)
```

Comment:
It feels a bit odd to me that plain ndarrays only use the `NdarrayMixin` if they are multi-field.  I realize it's unnecessary, but for the sake of consistency (as long as we're going wild with column mixins) would it hurt to use `NdarrayMixin` for all plain ndarray columns (i.e. the default)?

I'm just curious what the effect would be.